tests/ci_tests/code_formatter/crewai/test_crewai_agents.py

```diff
@@ -0,0 +1,310 @@
+        monkeypatch.delenv("COFFEE_MAKER_GEMINI_API_KEY", raising=False)
+
+        with pytest.raises(RuntimeError, match="Gemini API key missing"):
+            agents._resolve_gemini_api_key()
+
+
+class TestCreateCodeFormatterAgents:
+    """Tests for create_code_formatter_agents function"""
+
+    def test_returns_dict_with_senior_engineer(self):
+        """Test that the function returns a dict with 'senior_engineer' key"""
+        mock_langfuse_client = mock.MagicMock()
+
+        # Mock the prompts
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "Analyze and refactor code"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "You are a senior engineer with 10 years of experience"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        result = agents.create_code_formatter_agents(mock_langfuse_client)
+
+        assert isinstance(result, dict)
+        assert "senior_engineer" in result
+        assert isinstance(result["senior_engineer"], Agent)
+
+    def test_agent_has_correct_role(self):
+        """Test that the senior engineer agent has correct role"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal text"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory text"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        assert agent.role == "Senior Software Engineer"
+
+    def test_agent_uses_langfuse_prompts(self):
+        """Test that agent uses prompts from Langfuse"""
+        mock_langfuse_client = mock.MagicMock()
+
+        expected_goal = "Refactor code according to best practices"
+        expected_backstory = "You have extensive experience in Python development"
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = expected_goal
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = expected_backstory
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        # Verify the prompts were fetched
+        assert mock_langfuse_client.get_prompt.call_count == 2
+        mock_langfuse_client.get_prompt.assert_any_call("refactor_agent/goal_prompt")
```

Comment:
Added a docstring to the method, describing its purpose.
```suggestion
    def test_agent_has_no_tools(self):
```